src/Core/src/Credentials/Credentials.php

```diff
@@ -21,16 +23,20 @@ final class Credentials implements CredentialProvider
 
     private $expireDate;
 
+    private $cacheable;
+
     public function __construct(
         string $accessKeyId,
         string $secretKey,
         ?string $sessionToken = null,
-        ?\DateTimeImmutable $expireDate = null
+        ?\DateTimeImmutable $expireDate = null,
```

Comment:
Could we change the order of these? 
If the `Credentials` are not cacheable, then we wound not have an expiry date.

Reply:
thatwould be a BC break. I'm not sure it worth it

Reply:
Oh, Yes, You are correct. 

I thought both of these were new parameters.